frontend/packages/console-shared/src/components/spotlight/Spotlight.tsx

```diff
@@ -0,0 +1,68 @@
+import * as React from 'react';
+import cx from 'classnames';
+import Portal from '@console/shared/src/components/popper/Portal';
+import './Spotlight.scss';
+
+type SpotlightProps = {
+  selector: string;
+  animate?: boolean;
+  hasBackdrop?: boolean;
+};
+
+const isInViewport = (element) => {
+  const rect = element.getBoundingClientRect();
+  return (
+    rect.top >= 0 &&
+    rect.left >= 0 &&
+    rect.bottom <= (window.innerHeight || document.documentElement.clientHeight) &&
+    rect.right <= (window.innerWidth || document.documentElement.clientWidth)
+  );
+};
+
+const Spotlight: React.FC<SpotlightProps> = ({ selector, animate, hasBackdrop }) => {
+  const element = document.querySelector(selector);
+  const { height, width, top, left } = element.getBoundingClientRect();
+  const style: React.CSSProperties = {
+    top,
+    left,
+    height,
+    width,
+    ['--ocs-spotlight-top' as React.ReactText]: `${top}px`,
+    ['--ocs-spotlight-left' as React.ReactText]: `${left}px`,
+    ['--ocs-spotlight-height' as React.ReactText]: `${height}px`,
+    ['--ocs-spotlight-width' as React.ReactText]: `${width}px`,
+  };
+
+  const [clicked, setClicked] = React.useState(false);
+  React.useEffect(() => {
+    if (!isInViewport(element)) {
+      element.scrollIntoView();
+    }
+    const handleClick = () => setClicked(true);
+    document.addEventListener('mousedown', handleClick);
+    return () => {
+      document.removeEventListener('mousedown', handleClick);
+    };
```

Comment:
Only listen for clicks if in `interactive` mode.
You should also make sure the click occurs on the target element and not anywhere on screen.

Reply:
```suggestion
    const handleClick = () => setClicked(true);
    element.addEventListener('mousedown', handleClick);
    return () => {
      element.removeEventListener('mousedown', handleClick);
    };
```

Reply:
the goal for this function is to detect any click outside the `link` and stop the animation.

> The solid rectangle hint persists until the user clicks somewhere else on the UI or waits the 10 seconds until it fades away. 
although reading the uxd again, it seems the `click-anywhere-to-disappear` should only apply to the solid rectangle persisting after the expand animation. I'll check this with ux